cmds/boot/systemboot/main.go

```diff
@@ -60,6 +179,213 @@ func getSystemFWVersion(si *smbios.Info) (string, error) {
+	DIMMRequestDataParam4.ParameterSelector = 4
+	DIMMRequestDataParam5.ParameterSelector = 5
+	DIMMRequestDataParam6.ParameterSelector = 6
+
+	for index := 0; index < len(t17); index++ {
+		DIMMRequestDataParam1.Index = uint8(index)
+		DIMMRequestDataParam2.Index = uint8(index)
+		DIMMRequestDataParam3.Index = uint8(index)
+		DIMMRequestDataParam4.Index = uint8(index)
+		DIMMRequestDataParam5.Index = uint8(index)
+		DIMMRequestDataParam6.Index = uint8(index)
+
+		if t17[index].AssetTag == "NO DIMM" {
+			DIMMRequestDataParam1.DIMMPresent = 0xFF // 0xFF - Not Present
+		} else {
+			DIMMRequestDataParam1.DIMMPresent = 0x01 // 0x01 - Present
+		}
+
+		data := strings.Split(strings.TrimPrefix(t17[index].BankLocator, "_"), "_")
+		dimm, _ := strconv.ParseUint(strings.TrimPrefix(data[2], "Dimm"), 16, 8)
+		channel, _ := strconv.ParseUint(strings.TrimPrefix(data[1], "Channel"), 16, 8)
+		node, _ := strconv.ParseUint(strings.TrimPrefix(data[0], "Node"), 16, 8)
+		DIMMRequestDataParam1.DIMMNumber = uint8(dimm)
+		DIMMRequestDataParam1.ChannelNumber = uint8(channel)
+		DIMMRequestDataParam1.NodeNumber = uint8(node)
+
+		err = ipmi.SendOemIpmiCommand(_FB_OEM_IPMI_NET_FUNCTION, _FB_OEM_SET_DIMM_INFO, &DIMMRequestDataParam1)
+		if err != nil {
+			return err
+		}
+
+		if DIMMRequestDataParam1.DIMMPresent == 0x01 {
+			// DIMM type: bit[7:6] for DDR3 00-Normal Voltage(1.5V), 01-Ultra Low Voltage(1.25V), 10-Low Voltage(1.35V), 11-Reserved
+			//                     for DDR4 00~10-Reserved, 11-Normal Voltage(1.2V)
+			//            bit[5:0] 0x00=SDRAM, 0x01=DDR1 RAM, 0x02-Rambus, 0x03-DDR2 RAM, 0x04-FBDIMM, 0x05-DDR3 RAM, 0x06-DDR4 RAM
+			if t17[index].Type == smbios.MemoryDeviceTypeDDR3 {
+				switch t17[index].ConfiguredVoltage {
+				case 1500:
+					DIMMRequestDataParam2.DIMMType = 0x05
+				case 1250:
+					DIMMRequestDataParam2.DIMMType = 0x45
+				case 1350:
+					DIMMRequestDataParam2.DIMMType = 0x85
+				default:
+					DIMMRequestDataParam2.DIMMType = 0x05
+				}
+			} else {
+				switch t17[index].Type {
+				case smbios.MemoryDeviceTypeSDRAM:
+					DIMMRequestDataParam2.DIMMType = 0x00
+				case smbios.MemoryDeviceTypeDDR:
+					DIMMRequestDataParam2.DIMMType = 0x01
+				case smbios.MemoryDeviceTypeRDRAM:
+					DIMMRequestDataParam2.DIMMType = 0x02
+				case smbios.MemoryDeviceTypeDDR2:
+					DIMMRequestDataParam2.DIMMType = 0x03
+				case smbios.MemoryDeviceTypeDDR2FBDIMM:
+					DIMMRequestDataParam2.DIMMType = 0x04
+				case smbios.MemoryDeviceTypeDDR4:
+					DIMMRequestDataParam2.DIMMType = 0xC6
+				default:
+					DIMMRequestDataParam2.DIMMType = 0xC6
+				}
+			}
+			DIMMRequestDataParam3.DIMMSpeed = t17[index].Speed
+			DIMMRequestDataParam3.DIMMSize = uint32(t17[index].Size)
+			copy(DIMMRequestDataParam4.ModulePartNumber[:], t17[index].PartNumber)
+			data, _ := strconv.ParseUint(t17[index].SerialNumber, 16, 32)
+			DIMMRequestDataParam5.ModuleSerialNumber = uint32(data)
+			for i := 0; i < len(list); i++ {
```

Comment:
If the information is in a map you can avoid this for loop.

Reply:
pmazzini is right, do a map.

Reply:
Done.